fake_resolver.go

```diff
@@ -0,0 +1,166 @@
+// Fake DNS lookups, so that they always returned the specified
+// IP
+//
+// Inspired by the golang net/dnsclient_unix_test.go code
```

Comment:
This is going to show up as package-level GoDoc, not file level. You might want to move this comment to be on a struct or something.